src/vcpkg/base/files.cpp

```diff
@@ -2901,24 +2911,28 @@ namespace vcpkg
                 if (this_send_actual == -1)
                 {
                     ec.assign(errno, std::generic_category());
+                    // https://man7.org/linux/man-pages/man2/sendfile.2.html#NOTES
+                    // recommends fallback to read/write if sendfile returns EINVAL or ENOSYS
+                    if (errno == EINVAL || errno == ENOSYS)
+                    {
+                        lseek(source_fd.get(), offset, SEEK_SET);
+                        break;
+                    }
                     return false;
                 }
 
                 remaining_size -= this_send_actual;
             }
 
-            destination_fd.fchmod(source_stat.st_mode, ec);
-            return !ec;
-#elif defined(__APPLE__)
-            if (fcopyfile(source_fd.get(), destination_fd.get(), 0, COPYFILE_ALL) == -1)
+            if (!ec)
             {
-                ec.assign(errno, std::generic_category());
-                return false;
+                destination_fd.fchmod(source_stat.st_mode, ec);
+                return true;
             }
+            // Else fall back to read/write
+            ec.clear();
+#endif // ^^^ !(defined(__APPLE__) || defined(__linux__))
```

Comment:
`#endif  // ^^^ (defined(__linux__)`

Reply:
I have just submitted #544 to fix those.